### ISSUE TITLE
MissingRegionError when using assume_role_credentials

### ISSUE THREAD

**Issue by cgspohn** [fluent/fluent-plugin-s3#119]
When using assume_role_credentials, even though s3_region was defined in the configuration file, I got this error:

2015-11-05 01:50:19 +0000 [error]: fluent/engine.rb:172:rescue in run: unexpected error error_class=Aws::Errors::MissingRegionError error=#<Aws::Errors::MissingRegionError: missing region; use :region option or export region name to ENV['AWS_REGION']>

This was after upgrading to latest gem and it also picked up the latest aws sdk.

After editing:
/opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-s3-0.6.1/lib/fluent/plugin/out_s3.rb

And adding the :region option, the error was resolved.

**Comment by repeatedly** [comment]
Hmm... could you send a PR?
I'm not sure where is the code did you edit.

**Comment by cgspohn** [comment]
Hi,  for testing I added line 231 you see below.  I hard-coded the value just for testing purposes.   How do I send a PR?  

    227       when @assume_role_credentials
    228         c = @assume_role_credentials
    229         credentials_options[:role_arn] = c.role_arn
    230         credentials_options[:role_session_name] = c.role_session_name
    231         **credentials_options[:region] = 'us-east-1'**
    232         credentials_options[:policy] = c.policy if c.policy
    233         credentials_options[:duration_seconds] = c.duration_seconds if c.duration_seconds
    234         credentials_options[:external_id] = c.external_id if c.external_id
    235         options[:credentials] = Aws::AssumeRoleCredentials.new(credentials_options)

**Comment by cgspohn** [comment]
That was without the asterisks, sorry I was trying to make it bold.  For clarity, the code is, with line 231 being the addition with a hard-coded value for testing/debugging.

    227       when @assume_role_credentials
    228         c = @assume_role_credentials
    229         credentials_options[:role_arn] = c.role_arn
    230         credentials_options[:role_session_name] = c.role_session_name
    231         credentials_options[:region] = 'us-east-1'
    232         credentials_options[:policy] = c.policy if c.policy
    233         credentials_options[:duration_seconds] = c.duration_seconds if c.duration_seconds
    234         credentials_options[:external_id] = c.external_id if c.external_id
    235         options[:credentials] = Aws::AssumeRoleCredentials.new(credentials_options)

**Comment by repeatedly** [comment]
We have `s3_region` parameter so you cau use it.
See this code > https://github.com/fluent/fluent-plugin-s3/blob/d6b7097448445c847316ccee0fb6b57d58d473a9/lib/fluent/plugin/out_s3.rb#L107

**Comment by cgspohn** [comment]
So sorry, the code above obviously was wrong.  So when I did set the env variable AWS_REGION='us-east-1' the problem goes away.  I did have the s3_region setup in the config file, but looks like it totally ignored it when using assume role.

This crazy code change did indeed work when the environment variable was not setup.
options[:credentials] = Aws::AssumeRoleCredentials.new(client: Aws::STS::Client.new(region: 'us-east-1'), role_arn: c.role_arn, role_session_name: c.role_session_name)

Of course, would be better if the S3 plugin could use the s3_region parameter in assume role.
